src/tests/class_tests/openms/source/ExperimentalDesign_test.cpp

```diff
@@ -0,0 +1,316 @@
+  TEST_EQUAL(pl2f.size(), 24);
+  for (auto i : pl2f) { TEST_EQUAL((i.second >=1 && i.second <=3), true)}
+}
+END_SECTION
+
+START_SECTION((std::map< std::pair< String, unsigned >, unsigned> getPathLabelToFractionGroupMapping(bool) const ))
+{
+  // 12 quant. values from label-free, unfractionated files map to different fraction groups
+  std::map< std::pair< String, unsigned >, unsigned > pl2fg = labelfree_unfractionated_design.getPathLabelToFractionGroupMapping(true);
+  TEST_EQUAL(pl2fg.size(), 12);
+  int count = 1;
+  for (auto i : pl2fg) { TEST_EQUAL(i.second, count); ++count; }
+
+  pl2fg = fourplex_fractionated_design.getPathLabelToFractionGroupMapping(true);
+  TEST_EQUAL(pl2fg.size(), 24);
+  for (auto i : pl2fg) 
+  {
+    // extract fraction group from file name
+    int file(1); 
+    if (i.first.first.hasSubstring("TR2")) { file = 2; }
+    TEST_EQUAL(i.second, file); 
+  }
+}
+END_SECTION
+
+START_SECTION((unsigned getNumberOfSamples() const ))
+{
+  unsigned ns = labelfree_unfractionated_design.getNumberOfSamples();
+  TEST_EQUAL(ns, 12);
+
+  ns = fourplex_fractionated_design.getNumberOfSamples();
+  TEST_EQUAL(ns, 8);
+}
+END_SECTION
+
+START_SECTION((unsigned getNumberOfFractions() const ))
+{
+  unsigned nf = labelfree_unfractionated_design.getNumberOfFractions();
+  TEST_EQUAL(nf, 1);
+
+  nf = fourplex_fractionated_design.getNumberOfFractions();
+  TEST_EQUAL(nf, 3);
+}
+END_SECTION
+
+START_SECTION((unsigned getNumberOfLabels() const ))
+{
+  unsigned nl = labelfree_unfractionated_design.getNumberOfLabels();
+  TEST_EQUAL(nl, 1);
+
+  nl = fourplex_fractionated_design.getNumberOfLabels();
+  TEST_EQUAL(nl, 4);
+}
+END_SECTION
+
+START_SECTION((unsigned getNumberOfMSFiles() const ))
+{
+  unsigned nms = labelfree_unfractionated_design.getNumberOfMSFiles();
+  TEST_EQUAL(nms, 12);
+
+  nms = fourplex_fractionated_design.getNumberOfMSFiles();
+  TEST_EQUAL(nms, 6);
+}
+END_SECTION
+
+START_SECTION((unsigned getNumberOfFractionGroups() const ))
+{
+  unsigned nfg = labelfree_unfractionated_design.getNumberOfFractionGroups(); 
+  TEST_EQUAL(nfg, 12);
+
+  nfg = fourplex_fractionated_design.getNumberOfFractionGroups();
+  TEST_EQUAL(nfg, 2);
+}
+END_SECTION
+
+START_SECTION((unsigned getSample(unsigned fraction_group, unsigned label=1)))
+{
+  unsigned s = labelfree_unfractionated_design.getSample(1, 1); 
+  TEST_EQUAL(s, 1);
+  s = labelfree_unfractionated_design.getSample(12, 1);
+  TEST_EQUAL(s, 12);
+
+  s = fourplex_fractionated_design.getSample(1, 1);
+  TEST_EQUAL(s, 1);
+  s = fourplex_fractionated_design.getSample(2, 4);
+  TEST_EQUAL(s, 8);
+}
+END_SECTION
+
+START_SECTION((bool isFractionated() const ))
+{
+  bool b = labelfree_unfractionated_design.isFractionated(); 
+  TEST_EQUAL(b, false);
+
+  b = fourplex_fractionated_design.isFractionated(); 
+  TEST_EQUAL(b, true);
+}
+END_SECTION
+
+START_SECTION((bool sameNrOfMSFilesPerFraction() const ))
```

Comment:
make sure you test all public methods

Reply:
All are tested now, except the simple getter and setter of the inner struct and loading from identifications only data.